scripts/coccinelle/reserved_function_names.cocci

```diff
@@ -0,0 +1,98 @@
+/// Find function names starting with '__' or '_' and rename them
+// Copyright: (C) 2018 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+virtual patch
+virtual report
+
+@r_anyfn disable optional_storage@
+identifier fn =~ "^_.*";
+position p;
+@@
+
+  fn@p (...)
+
+@r_staticfn disable optional_storage@
+identifier fn =~ "^_.*";
+position p;
+@@
+
+  static fn@p (...) {...}
+
+@script:python depends on report && !r_staticfn@
+fn << r_anyfn.fn;
+pos << r_anyfn.p;
+@@
+msg="WARNING: Reserved function name " + fn
+coccilib.report.print_report(pos[0], msg)
+
+@script:python depends on report@
+fn << r_staticfn.fn;
+pos << r_staticfn.p;
+@@
+msg="WARNING: Reserved static function name " + fn
+coccilib.report.print_report(pos[0], msg)
+
+@script:python r_static_rewrite depends on patch@
+fn << r_staticfn.fn;
+z;
+@@
+
+r = fn
+
+while r.startswith('_'):
+    r = r[1:]
+
+coccinelle.z = r
+
+@script:python r_any_rewrite depends on patch@
+fn << r_anyfn.fn;
+z;
+@@
+
+prefixes = { 'arch': 'z_arch_',
+             'sys': 'z_sys_',
+	     'k': 'z_',
+	     'impl': 'z_sys_impl_',
+	     'handler': 'z_sys_handl_',
+
+	     'current': 'current',
+	     'ready_q': 'ready_q',
+	     'timeout_q': 'timeout_q',
+	     'threads': 'threads' }
+
+r = fn
+
+while r.startswith('_'):
+    r = r[1:]
+
+for p in prefixes:
+    if r.startswith(p):
+        r = r[len(p):]
+
+	while r.startswith('_'):
+	    r = r[1:]
+
+	r = prefixes[p] + r
+	break
+
+else:
+    r = 'z_' + r
+
+coccinelle.z = r
+
+@r_static_out depends on patch@
+identifier r_staticfn.fn;
+identifier r_static_rewrite.z;
```

Comment:
`identifier  r_staticfn.fn, r_static_rewrite.z;` is fine.

Reply:
Ok, let's go with those once we're experts. Right now my head is spinning less if I have them on separate lines - go figure... :-)